conda_forge_tick/contexts.py

```diff
@@ -27,10 +29,10 @@ class MigratorSessionContext:
     dry_run: bool = True
 
 
-@dataclass
+@dataclass(frozen=True)

```

Comment:
In particular this pr appears to set _drfault_branch below which is odd in python since that is formally a private attribute.

Reply:
That's great! I couldn't figure out how to use post init with a frozen dataclass. Thanks!

Reply:
Okay, sadly this had one more flaw: We want the type annotation of default_branch to be `str` and not `str | None` because we ensure str in post_init.

Because dataclass attributes cannot have different type annotations in `__init__` than as attribute, I had to introduce an init var in 32fc983.

This is the best solution I could come up with that avoids `# type: ignore` annotations while still having default_branch typed as str.

Reply:
We should set the default to `""` instead of `None` and use the simpler code.

Reply:
Sounds good. Here we go: d03f0f1